app/src/main/java/chat/rocket/android/activity/ServerConfigActivity.java

```diff
@@ -77,14 +83,27 @@ protected void onPause() {
   }
 
   private void onRenderServerConfigError(ServerConfig config) {
-    if (config != null) {
+    if (config.getState() == ServerConfig.STATE_CONNECTION_ERROR) {
       sessionObserver.unsub();
       showFragment(new RetryConnectFragment());
     } else {
       sessionObserver.sub();
     }
   }
 
+  private void continueWithServerConfigSession(final Session session) {
+    fetchPublicSettings()
+        .continueWith(task -> {
+          registerForPush();
+          return task;
+        })
+        .continueWith(task -> {
+          onRenderServerConfigSession(session);
+          return task;
+        })
+        .continueWith(new LogcatIfError());
+  }
```

Comment:
Do the `SessionObserver#onLogin` runs before the user logs in? If it does, that looks fine.

If not, we could try something else. The idea is to run those calls before the user login.